pkg/cableengine/healthchecker/pinger.go

```diff
@@ -111,9 +111,17 @@ func (p *pingerInfo) doPing() error {
 		if pinger.PacketsSent-pinger.PacketsRecv > int(p.maxPacketLossCount) {
 			p.Lock()
 			defer p.Unlock()
+			// The counter will be reset to once to ensure that it is continuous packet loss
+			// before marking the connection as down.
+			if pinger.PacketsSent <= int(p.maxPacketLossCount)+1 {
```

Comment:
So basically it seems we want to mark it down if we we've sent out `maxPacketLossCount + 1` packets with no replies.  I think it would be clearer to check:

```suggestion
			if pinger.PacketsRecv == 0  {
```

Either way once we trip the threshold it will take another `maxPacketLossCount + 1` packets lost before we mark it as down (unless the very first `maxPacketLossCount + 1` packets happened failed). So it will take `(maxPacketLossCount + 1)*2` total packets lost before we mark it as down. 

As an alternative, we could reset `PacketsSent` and `PacketsRecv` when we get a reply in `OnRecv`.  Then here we would do:

```
if pinger.PacketsSent-pinger.PacketsRecv > int(p.maxPacketLossCount) {
    p.Lock()
    defer p.Unlock()

    p.failureMsg = fmt.Sprintf("Failed to successfully ping the remote endpoint IP %q", p.ip)
    p.connectionStatus = ConnectionError

    pinger.Stop()
}
```

Reply:
Done.